src/overlays/actors/ovl_En_Dnp/z_en_dnp.c

```diff
@@ -22,61 +65,454 @@ const ActorInit En_Dnp_InitVars = {
-static CollisionCheckInfoInit2 D_80B3DCAC = { 0, 0, 0, 0, MASS_IMMOVABLE };
+    if ((this->unk_32E != 0) && (func_80152498(&globalCtx->msgCtx) == 2)) {
+        Actor_MarkForDeath(&this->actor);
+    } else if (this->unk_32E == 0) {
+        if (func_800B84D0(&this->actor, globalCtx)) {
+            this->unk_32E = 1;
+        } else {
+            this->actor.textId = 0x971;
+            player->actor.textId = this->actor.textId;
+            func_800B8500(&this->actor, globalCtx, 9999.9f, 9999.9f, -1);
+        }
+    }
+}
 
+void func_80B3D3F8(EnDnp* this, GlobalContext* globalCtx) {
+    if (func_8010BF58(&this->actor, globalCtx, D_80B3DE58, 0, &this->unk_328)) {
+        func_8013AED4(&this->unk_322, 3, 7);
+        this->unk_322 &= ~0x8;
+        this->actionFunc = func_80B3D2D4;
+    } else {
+        Math_ApproachS(&this->actor.shape.rot.y, this->actor.yawTowardsPlayer, 3, 10920);
+    }
+}
 
-extern ColliderCylinderInit D_80B3DC80;
-extern CollisionCheckInfoInit2 D_80B3DCAC;
-#endif
+void func_80B3D47C(EnDnp* this, GlobalContext* globalCtx) {
+    if (this->actor.bgCheckFlags & 1) {
+        Math_SmoothStepToF(&this->actor.scale.x, 0.0085f, 0.1f, 0.01f, 0.001f);
+        if ((s32)(this->actor.scale.x * 10000.0f) >= 85) {
+            this->actor.flags |= 1;
+            func_8013AED4(&this->unk_322, 3, 7);
+            this->unk_322 &= ~0x10;
+            this->unk_322 |= 0x400;
+            this->actor.scale.x = 0.0085f;
+            this->actionFunc = func_80B3D558;
+        }
+    }
+    Actor_SetScale(&this->actor, this->actor.scale.x);
+}
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Dnp_0x80B3CA20/func_80B3CA20.asm")
+void func_80B3D558(EnDnp* this, GlobalContext* globalCtx) {
+    if (ActorCutscene_GetCanPlayNext(this->actor.cutscene)) {
+        ActorCutscene_StartAndSetUnkLinkFields(this->actor.cutscene, &this->actor);
+        gSaveContext.weekEventReg[23] |= 0x20;
+    } else {
+        ActorCutscene_SetIntentToPlay(this->actor.cutscene);
+    }
+}
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Dnp_0x80B3CA20/func_80B3CC38.asm")
+void EnDnp_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnDnp* this = THIS;
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Dnp_0x80B3CA20/func_80B3CC80.asm")
+    ActorShape_Init(&this->actor.shape, 0.0f, func_800B3FC0, 16.0f);
+    SkelAnime_InitSV(globalCtx, &this->skelAnime, &D_06010D60, NULL, this->jointTable, this->morphTable, 26);
+    this->unk_340 = -1;
+    func_80B3CC38(this, 15);
+    Collider_InitAndSetCylinder(globalCtx, &this->collider, &this->actor, &sCylinderInit);
+    CollisionCheck_SetInfo2(&this->actor.colChkInfo, DamageTable_Get(0x16), &sColChkInfoInit);
+    this->unk_322 = 0;
+    this->actor.targetMode = 0;
+    this->unk_322 |= (0x100 | 0x80 | 0x10);
+    this->actor.gravity = -1.0f;
+    if (ENDNP_PARAMS_7(&this->actor) == ENDNP_PARAMS_7_1) {
+        this->actor.flags &= -2;
+        Actor_SetScale(&this->actor, 0.00085000007f);
```

Comment:
Is the precision here necessary? Just wondering since later in this function there's a set scale with 0.0085f and it's OK

Reply:
Turns out that yeah, it does. 0x3A5ED28A vs 0x3A5ED289

Reply:
Maybe it's `17.0f / 20000.0f`

Reply:
Doesn't match either.